File: ArraysAndHashing/Easy/448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.kt
class Solution {
    fun findDisappearedNumbers(nums: IntArray): List<Int> {
        val list: MutableList<Int> = MutableList(nums.size) { 0 }
        val output = mutableListOf<Int>()

        nums.forEach {
            // println(it)
            // println(it - 1)
            // println()
            list[it - 1] = list[it - 1] + 1 ?: 1
        }

        println(list)

        list.forEachIndexed { index, `val` ->
            if (`val` == 0) {
                output.add(index + 1)   
            }
        }
        
        println(output)
        return output
    }
}

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.ts
function findDisappearedNumbers(nums: number[]): number[] {
    /**
    create set with nums
    loop from 1 .. nums.size
        if current index not in set:
            add to []

    return []
     */
    let numSet = new Set(nums);
    let output: number[] = [];
    for (let i = 1; i <= nums.length; i++) {
        if (!numSet.has(i)) {
            output.push(i)
        }
    }

    return output;
};